src/query_plan/operation.rs

```diff
@@ -406,6 +437,9 @@ fn merge_selections(
                             }
                         }
                     }
+                    NormalizedSelection::NormalizedFragmentSpread(_) => {
+                        // nothing to do fragment spread is already part of the selection set
```

Comment:
nit: Might be nice to copy over [the relevant comment from Sylvain here](https://github.com/apollographql/federation/blob/f69a0694b95e99c22f0d5f7bc6df1b46568b2a8d/internals-js/src/operations.ts#L2182-L2185).